### ISSUE TITLE
Linter stopped working unexpectedly.

### ISSUE THREAD

**Issue by Hadkins2** [halcyon-tech/vscode-rpgle#67]
Linter was working two days ago and stopped unexpectedly.  I checked all output items for errors and couldn't find any issues reported.

Today I uninstalled all of the Code for IBM i extensions and reinstalled.  No effect.

Finally found when I hovered over the gear to to get the version of RPGLE language tools to report this, 13 of these messages:

Activation time : 4ms
Uncaught Errors (13)
Cannot read property 'procedures' of undefined

My RPGLINT.JSON file.  FYI, I deleted it and let the program recreate it.
{
  "BlankStructNamesCheck": true,
  "QualifiedCheck": true,
  "PrototypeCheck": true,
  "ForceOptionalParens": true,
  "NoOCCURS": true,
  "NoSELECTAll": true,
  "UppercaseConstants": true,
  "IncorrectVariableCase": true,
  "StringLiteralDupe": true,
  "NoSQLJoins": true,
  "NoCTDATA": true,
  "PrettyComments": true,
  "NoGlobalSubroutines": true,
  "NoLocalSubroutines": true,
  "UppercaseDirectives": true
}

version v0.6.1 for the RPGLE Language tools extension
version v0.9.1 for Code for IBM i extension

Thanks,

Howard

**Comment by worksofliam** [comment]
Strange. So it crashes even before you open any code? Does it crash when you open a specific member? Are you able to share the code it crashed with?

**Comment by Hadkins2** [comment]
I think I may have found the issue.  But it is weird that it worked before.  If I comment out "/COPY" code that is not written as **FREE, the linter kicks in with no problems.  So I guess I just need to convert the copy code and all will be right in the world.

Thanks for prompting me to dig deeper.  I gave up trying to figure it out too soon.

Howard

**Comment by worksofliam** [comment]
Strange! The linter should continue working even if you have a copybook that is not free format. I will see if I can find something out too later.

**Comment by Hadkins2** [comment]
Well now I am completely confused.  I put the copy books back in one at a time just to test which one kills the process.  The linter continued to work, but it is getting the uncaught exceptions again.

**Comment by worksofliam** [comment]
@Hadkins2 Would you be willing to share the code? Sharing privately is an option too.

**Comment by Hadkins2** [comment]
I think I have found the issue.

In each of the copybooks, it appears the original programmer included comments in the copybook code, but didn't want to that carry into the source code of the program using it and used the /EOF compiler directive to make it stop reading the copybook.

Here is the offending copybook.

```
D UPPERCASE       PR          4096    Varying                            
D   String                    4096    Const Varying                      
D   Escaped                       n   Const Options(*NoPass)             
 /EoF                                                                    
       Converts all of the letters in String to their                    
       UPPER CASE equivalents.  Non-alphabetic characters                
       remain unchanged.                                                 
                                                                         
       Escaped = *ON = converts characters that would crash iPDF and     
                       HTML to approximately equivalent characters.      
                       For example, translate " and ' to ` .             
                       (Default)                                         
                 *OFF= Do not convert any characters other than A-Z.     
                                                                         
H BNDDIR('WVWEB/COMMON')                                                 
 /Copy WVPROT/format,UPPERCASE                                           
                                                                         
```

Some of the other copy books have /IF directives in them as well.  I put an asterisk in column 7 in this copybook and the linter was happy.

Howard

**Comment by worksofliam** [comment]
@Hadkins2 Ha, great find! I will work on the `/EOF` support right now.

**Comment by worksofliam** [comment]
Fix coming out in just a few minutes :)

**Comment by worksofliam** [comment]
@Hadkins2 GIve it a go in 0.6.2 and let me know.

**Comment by worksofliam** [comment]
Hey @Hadkins2

Did you get to try this out yet?

Thanks!

**Comment by Hadkins2** [comment]
Apologies for the delay.  I did get the update the next morning and it didn't resolve the issue.  There has to be something wrong with the copybooks and not the linter.  I made copies of the copybooks and removed everything but the actual code and it still was giving me issues.  I got in a sort of a time crunch and couldn't spend more time on it now.  I resolved my issue by copying the code from the two offending copybooks into the copybook I use for Data Structure Templates in this program (a nice tip I learned from your YT videos) and converted them to **FREE.  They are working just fine now.  I will circle back to this once I have met the deadline on this project.

I will keep you updated.

Howard
